### PR TITLE
Basic parameter server example.

### PR DESCRIPTION
**Pull request by robertnishihara**
I'd like to get people's feedback on this. This takes a basic convnet for MNIST and trains it using a parameter server (using distributed actor handles).

The parameter server implements a very simple push/pull API (easily extendable to have a custom "updater") using just a dictionary of numpy arrays.

Each worker is a "task" implemented using a TensorFlow model, which calls the parameter server's push/pull API.

Note it should be simple to provide a "sharded parameter server" abstraction which splits the parameter server's weights across multiple actors.

cc @pcmoritz @ericl @rshin @elibol @Wapaul1

### CODE REVIEW

File: examples/parameter_server/model.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+            self.variables = ray.experimental.TensorFlowVariables(
+                self.cross_entropy, self.sess)
+
+            self.grads = self.optimizer.compute_gradients(
+                self.cross_entropy)
+
+    def compute_update(self, x, y):
+        # TODO(rkn): Computing the weights before and after the training step
+        # and taking the diff is awful.
```
Comment (robertnishihara) [inline examples/parameter_server/model.py]: Note this comment, this is completely unacceptable.

**Comment by ericl** [comment]
Interesting, do you have an estimate of the overhead here compared to distributed tensorflow? e.g. at which point do we saturate the single-threaded actor

File: examples/parameter_server/model.py (hunk 1 above)
Comment (richardliaw) [reply examples/parameter_server/model.py]: one option is to do this in tf land instead of doing this copy out two times like a `self.sess.run(self.delta)` where `self.delta` does a `old_weights = tf.identity(v1), self.train_step, self.delta = new_weights - oldweights`

Comment (richardliaw) [reply examples/parameter_server/model.py]: might not help that much

**Comment by robertnishihara** [comment]
cc @wanglin-bjhl

**Comment by robertnishihara** [comment]
@ericl Good question, haven't benchmarked it yet.

**Comment by robertnishihara** [comment]
On a single machine, if I speed up the worker in order to just benchmark the parameter server, it seems to top out at around 90 ops/s (using MNIST weights). If I make the weights negligible in size, then it seems to top out around 1000 ops/s.

**Review by pcmoritz — APPROVED** [review]

File: examples/parameter_server/model.py (hunk 1 above)
Comment (elibol) [reply examples/parameter_server/model.py]: This also seems like an approach worth looking into: https://stackoverflow.com/questions/34687761/efficiently-grab-gradients-from-tensorflow
